### ISSUE TITLE
Encrypted file detection module needed

### ISSUE THREAD

**Issue by bcarrier** [sleuthkit/autopsy#207]
We should have an ingest module that detects encrypted files.  Detected files should have a TSK_ENCRYPTION_DETECTED attribute added to the blackboard.

**Comment by xQsme** [comment]
I know this is an old issue, but I was analyzing the Autopsy source code and found the encryption module very basic since it only focuses on file entropy.
I'm a cybersecurity master's student and one of the projects I developed was encryption detection, the repository can be found here: https://github.com/xQsme/cryptoFileFinder
This project is mostly a proof of concept...
Firstly calculating file entropy allows a quick way to disregard a lot of files without too much computing power.
Following that with the calculation of the chi-squared of byte distribution and comparing it with a limit calculated based on file size allows disregarding a lot of seemingly random files like images.
Finally I found out that calculating the chi squared for byte sequences of size 2 and again comparing it with a limit based on file size allowed a pretty accurate encryption finding algorithm.
We had few false positives and zero false negatives, but we only tested 4 different encryption algorithms (AES, DES, 3DES and BF) and applied each to files of 500 bytes, 5KB, 50KB, 500KB, 5MB, 50MB and 300MB
These limits I talked about should be adjusted as more encryption algorithms are tested to make sure every encrypted file is caught by the algorithm.
This is merely a suggestion to further develop the encryption module, I'm available to answer any questions if needed.
